packages/compass/src/main/menu.ts

```diff
@@ -452,6 +452,14 @@ class CompassMenu {
       this.refreshMenu();
     });
 
+    preferences.onPreferenceValueChanged('readOnly', newReadOnly => {
+      track('Read-Only Changed', {
+        readOnly: newReadOnly
+      });
```

Comment:
Might be good to double-check with product that it makes sense to track this event